### DISCUSSION TITLE
Need documentation for `@endid` on trills

### DISCUSSION THREAD

**Discussion by th-we** [music-encoding/music-encoding#1495] [Q&A] [answered]
I'm not clear about whether the `@endid` of a trill should point to the last note of the trill or the first note/rest/whatever after the trill. Verovio seems to interpret `@endid` as the first thing after the trill:

    <measure n="1">
        <staff n="1">
            <layer n="1">
                <note xml:id="m-38" dur="1" oct="4" pname="g"/>
            </layer>
        </staff>
        <trill layer="1" staff="1" startid="#m-38" endid="#m-43"/>
    </measure>
    <measure n="2">
        <staff n="1">
            <layer n="1">
                <note xml:id="m-43" dur="1" oct="4" pname="g"/>
            </layer>
        </staff>
    </measure>

![grafik](https://user-images.githubusercontent.com/1147152/73738581-a16b6000-473c-11ea-8dd4-1e28327ae2f5.png)

Is that the recommended encoding?  At the moment, there are only `@tstamp2` examples in the documentation.

**Comment by craigsapp** [comment]
I use `@tstamp2` in such cases (I think I use `@tstamp2` in all cases for trill lines):

<img width="1582" alt="Screen Shot 2020-02-04 at 11 41 45" src="https://user-images.githubusercontent.com/3487289/73780454-5c96f780-4743-11ea-98a5-f29cc1f77dc0.png">

using `@endid` for the next note would be somewhat problematic since the trill line should not cross the barline, that is why I added a tstamp that mean the end of the measure instead.

**Comment by rettinghaus** [comment, marked as answer]
Usually `@endid` points to the last constituent of the affected ones, so the trill should last until the second note in the example. (Like a glissando has a start note and an end note, but the glissando doesn't continue on the end note.)
Trills to bar lines should (in this particular case) use `@tstamp2="5"`.

**Comment by musicEnfanthen** [comment]
Transferring to main repo

**Comment by musicEnfanthen** [comment]
Converting to discussion

**Comment by musicEnfanthen** [comment]
Closing discussion for now.
